engine/consensus/matching/engine.go

```diff
@@ -209,60 +207,50 @@ func (e *Engine) onReceipt(originID flow.Identifier, receipt *flow.ExecutionRece
 		Logger()
 
 	resultFinalState, ok := receipt.ExecutionResult.FinalStateCommitment()
-	if !ok {
-		return fmt.Errorf("could not get final state: no chunks found")
+	if !ok || len(resultFinalState) < 1 {
```

Comment:
nice suggestion Leo. Extended goDoc for `ExecutionResult` to be clear about protocol convention. 

On a related note:  was thinking that maybe instead of a `bool` we should return an `error`
https://github.com/onflow/flow-go/blob/69f4fd8fc8402bec589cf061ef09a6c493b28dbd/model/flow/execution_result.go#L36-L37

what do you think